src/core/plugin.cpp

```diff
@@ -27,18 +28,76 @@ std::shared_ptr<config::section_t> wf::config_backend_t::get_output_section(
     return config.get_section(name);
 }
 
+static struct udev_property_and_desc
+{
+    char const *property_name;
+    char const *description;
+} properties_and_descs[] =
+{
+    {"ID_PATH", "stable physical connection path"},
+    {"ID_SERIAL", "stable vendor+pn+sn info"},
+    {"LIBINPUT_DEVICE_GROUP", "stable libinput info"},
+    // sometimes it contains info "by path", sometimes "by id"
+    {"DEVPATH", "unstable devpath"},
+    // used for debugging, to find DEVPATH and match the right
+    // device in `udevadm info --tree`
+};
+
 std::shared_ptr<config::section_t> wf::config_backend_t::get_input_device_section(
     wlr_input_device *device)
 {
+    auto& config = wf::get_core().config;
+    std::shared_ptr<wf::config::section_t> section;
+    auto print_devpath = getenv("WF_PRINT_UDEV_DEVPATH");
+
+    if (wlr_input_device_is_libinput(device))
+    {
+        auto libinput_dev = wlr_libinput_get_device_handle(device);
+        if (libinput_dev)
+        {
+            udev_device *udev_dev = libinput_device_get_udev_device(libinput_dev);
+            if (udev_dev)
+            {
+                for (struct udev_property_and_desc const & pd : properties_and_descs)
+                {
+                    if (!print_devpath && !strncmp(pd.property_name, "DEVPATH", strlen("DEVPATH")))
```

Comment:
Actually, wouldn't it be more appropriate to send this info over via IPC and print it in the scripts we already have for listing input devices? I can imagine that this would be much easier to use.

Otherwise, I think we can/should just print every property we are querying here (in the corresponding debug category).

Reply:
I disagree that we should require ipc for this task. It should be as easy as `wayfire -d input-device` (or plural, input-devices).